app/Services/HomeService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Product;
+use App\Models\Category;
+use App\Models\ProductImage;
+use App\Utilities\GoogleDriveUtility;
+
+
+class HomeService
+{
+    private $googleDriveUtility;
+    /**
+     * Summary of __construct
+     */
+    public function __construct() 
+    {
+        $this->googleDriveUtility = new GoogleDriveUtility();
+    }
+
+    /**
+     * Summary of getHomeProduct
+     * @return mixed
+     */
+    public function getHomeProduct()
+    {
+        //hitung product kelipatan 4
+        $totalProducts = Product::count();
+        $limit = $totalProducts - ($totalProducts%4);
+        if($limit == 0 && $totalProducts >0){
+            $limit = $totalProducts;
+        }
+        
+        $products = Product::take($limit)->get()
+        ->map(function($product) {
```

Comment:
`$products = Product::take($limit)->get()
            ->map(function($product) {
                $name = $product->name;
                $price = $product->price;
                $rating = $product->sustainability_score;

                $imgUrl = ProductImage::where('product_id', $product->id)->first();
                $img = $this->googleDriveUtility->getImage($imgUrl->url);
                
                $link = route('getProduct', base64_encode("$name-$product->id"));

                return (object) compact('name', 'rating', 'price', 'img', 'link');
            });`
            
Space ny benerin jdi gitu

Reply:
resolved